kubernetes.go

```diff
@@ -320,13 +330,13 @@ func (c *KubernetesClient) Watch() error {
 			c.eventForwarder(e)
 		}
 
-		if !more {
+		if !more || e.Type == watch.Error {
```

Comment:
What I have seen is that if we receive an error, the channel is closed, so for next iteration `more` will be `false` and then we'd reconnect anyway.
We could just wait for the next iteration, but we are reading from 3 channels, if after a reconnection we are requesting an incorrect version, we'll receive an error from the 3 channels and then we could be interating on errors up to three times, depending on what channel is selected on each iteration. The (not so big) problem with that is that then we'll be handling the error from one to three times when we know that we have to reconnect at the end. So I decided to reconnect directly on the first error.

This `Watch` method is getting quite big and complex, we should probably refactor it, we could for example handle each watcher separatedly.